modules/dnn/src/model.cpp

```diff
@@ -0,0 +1,210 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+// Copyright (C) 2018-2019, Intel Corporation, all rights reserved.
+// Third party copyrights are property of their respective owners.
+
+
+#include "precomp.hpp"
+#include <algorithm>
+#include <iostream>
+#include <utility>
+#include <iterator>
+
+#include <opencv2/imgproc.hpp>
+
+namespace cv {
+namespace dnn {
+
+struct Model::Impl {
+    Size   size;
+    Scalar mean;
+    float  scale = 1.0;
+    bool   swapRB = true;
+    bool   crop = false;
+};
+
+Model::Model(const std::string& model, const std::string& config)
+    : Net(readNet(model, config)), impl_(new Impl) {};
+
+Model::Model(const Net& network) : Net(network), impl_(new Impl) {};
+
+Model& Model::setInputSize(const Size& size) {
+    impl_->size = size;
+    return *this;
+}
+
+Model& Model::setInputSize(int height, int width) {
+    impl_->size = Size(width, height);
+    return *this;
+}
+
+Model& Model::setInputMean(const Scalar& mean) {
+    impl_->mean = mean;
+    return *this;
+}
+
+Model& Model::setInputScale(float scale) {
+    impl_->scale = scale;
+    return *this;
+}
+
+Model& Model::setInputCrop(bool crop) {
+    impl_->crop = crop;
+    return *this;
+}
+
+Model& Model::setInputSwapRB(bool swapRB) {
+    impl_->swapRB = swapRB;
+    return *this;
+}
+
+std::pair<int, float> Model::classify(InputArray frame) {
+    Mat blob;
+    blobFromImage(frame, blob, 1.0, impl_->size, Scalar(), impl_->swapRB, impl_->crop, CV_8U);
+    setInput(blob, "", impl_->scale, impl_->mean);
+    Mat out = forward();
+
+    double conf;
+    cv::Point maxLoc;
+    minMaxLoc(out, nullptr, &conf, nullptr, &maxLoc);
+
+    return {maxLoc.x, static_cast<float>(conf)};
+}
+
+std::vector<Mat> Model::predict(const Mat& frame) {
+    Mat blob;
+    blobFromImage(frame, blob, 1.0, impl_->size, Scalar(), impl_->swapRB, impl_->crop, CV_8U);
+    setInput(blob, "", impl_->scale, impl_->mean);
+
+    std::vector<String> outNames = getUnconnectedOutLayersNames();
+    std::vector<Mat> outs;
+    forward(outs, outNames);
+    return outs;
+}
```

Comment:
`blob` and `outNames` could be a part of `Impl`. BTW, what do you think about moving an entire method to Impl?

```cpp
std::vector<Mat> Model::predict(const Mat& frame) {
    impl->predict(frame);
}
```